### PR TITLE
[backend/frontend] Fix import

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Related issues

* https://github.com/OpenBAS-Platform/openbas/issues/1067

### CODE REVIEW

**Comment by Dimfacion** [comment]
[Probleme de Prod_2024-06-13T14_32_02.159342300Z_(with_teams & with_players & with_variable_values).zip](https://github.com/user-attachments/files/15822109/Probleme.de.Prod_2024-06-13T14_32_02.159342300Z_.with_teams.with_players.with_variable_values.zip)
As seen together, I've had NPE with this import zip

File: openbas-model/src/main/java/io/openbas/database/repository/TagRepository.java [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 
-    @NotNull
-    Optional<Tag> findById(@NotNull String id);
+  @NotNull
+  Optional<Tag> findById(@NotNull String id);
 
-    Optional<Tag> findByName(@NotNull String name);
+  @NotNull
+  Optional<Tag> findByName(@NotNull final String name);
+
```
Comment (savacano28) [inline openbas-model/src/main/java/io/openbas/database/repository/TagRepository.java:20]: Maybe we could also add an ignoreCase here? Do we have a unique constraint on this column?

Comment (RomuDeuxfois) [reply openbas-model/src/main/java/io/openbas/database/repository/TagRepository.java:20]: We already have it :)

  @NotNull
  List<Tag> findByNameIgnoreCase(@NotNull final String name);

**Comment by RomuDeuxfois** [comment]
> [Probleme de Prod_2024-06-13T14_32_02.159342300Z_(with_teams & with_players & with_variable_values).zip](https://github.com/user-attachments/files/15822109/Probleme.de.Prod_2024-06-13T14_32_02.159342300Z_.with_teams.with_players.with_variable_values.zip) As seen together, I've had NPE with this import zip



> [Probleme de Prod_2024-06-13T14_32_02.159342300Z_(with_teams & with_players & with_variable_values).zip](https://github.com/user-attachments/files/15822109/Probleme.de.Prod_2024-06-13T14_32_02.159342300Z_.with_teams.with_players.with_variable_values.zip) As seen together, I've had NPE with this import zip

It's fixed, and it revealed another bug.
We no longer link the documents to the exercise when creating an inject in an exercise.

**Review by Dimfacion — APPROVED** [review]

**Pull request merged**